Library/Formula/z3.rb

```diff
@@ -6,6 +6,9 @@ class Z3 < Formula
   head "https://github.com/Z3Prover/z3.git"
   revision 1
 
+  depends_on :python => :recommended
```

Comment:
If you're going to make this recommended it needs to use the system Python by default.

```ruby
depends_on :python => :recommended if MacOS.version <= :snow_leopard
```